### PR TITLE
feat: ✨ finalizes collapsed tweet widget

### PR DESCRIPTION
**Pull request by Roaa94**
## Status

**READY**

## Description

This finalizes the `CollapsedTweet` widget/component along with other features necessary to do so. For example:
* Adding active state to the `Username` and `DisplayName` components with hover underline text effect and clickability.
   * Note: to make the Hover effect reusable (as it's used in many places across the app), the `HoverDetector` widget was created. It has a `childBuilder` with access to a `isHovered` variable. 

* Features added to the `CollapsedTweet` widget:
   * The vertical line bellow the avatar
   * The `Show this thread` section
   * The `Show more replies` section
   * Hover effect

Other additions in this PR:
* Renames `user` to `author` in `Tweet` model. This is to correspond to Twitter API's `author_id` param.

⚠️ Note:
The structure of the `Tweet` model and how the `CollapsedTweet` widget is configured will keep changing as the implementation gets closer to using a data layer.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/core/lib/src/widgets/tweet/collapsed_tweet.dart [hunk 1]
```diff
@@ -8,50 +7,176 @@ class CollapsedTweet extends StatelessWidget {
...
 
   /// The tweet data
   final Tweet tweet;
 
+  /// Whether this tweet is part of a thread
+  ///
+  /// If `true`, a vertical line bellow the avatar appears
+  final bool isThread;
+
+  /// OnPressed callback for `Show this thread` text
```
Comment (jenshor) [inline packages/core/lib/src/widgets/tweet/collapsed_tweet.dart:23]: This looks amazing, great job 💪🏻

File: app_widgetbook/lib/core/user/display_name.dart [hunk 2]
```diff
@@ -6,10 +6,29 @@ import 'package:widgetbook_annotation/widgetbook_annotation.dart';
...
+      displayName: context.knobs.text(
+        label: 'Value',
+        description: 'Enter display name',
+        initialValue: 'John Doe',
+      ),
+      id: '',
+      username: '',
     ),
   );
 }
```
Comment (jenshor) [inline app_widgetbook/lib/core/user/display_name.dart:34]: What's the difference between these two use-cases?

Comment (jenshor) [reply app_widgetbook/lib/core/user/display_name.dart:34]: Ok figured it out. It's the hover effect. Would be great to have an option within widgetbook to enable hover, but I'm not exactly sure how we can enable such functionality.

**Review by jenshor — APPROVED** [review]
The tweet component looks amazing 🎉

File: app_widgetbook/lib/core/user/display_name.dart (hunk 2 above)
Comment (Roaa94) [reply app_widgetbook/lib/core/user/display_name.dart:34]: Yes that's a great idea. 
For this specific component I think we'd have to expose the `isHovered` variable. But I think you mean have a general `Hovered` property which might be achieved in some way with a `MouseRegion` widget around the frame. I'll add it to the features ideas list 👍

**Pull request merged**